initialization/initialization_test.go

```diff
@@ -834,6 +834,45 @@ func TestStop(t *testing.T) {
 	}
 }
 
+func TestValidateComputeBatchSize(t *testing.T) {
+	cfg := config.DefaultConfig()
+	opts := config.DefaultInitOpts()
+
+	// Set invalid value of 0
+	opts.ComputeBatchSize = 0
+
+	_, err := NewInitializer(
+		WithNodeId(nodeId),
+		WithCommitmentAtxId(commitmentAtxId),
+		WithConfig(cfg),
+		WithInitOpts(opts),
+		WithLogger(testLogger{t: t}),
+	)
+	assert.Error(t, err)
```

Comment:
It's good to verify that the expected error happened. You can do it with `assert.ErrorContains`

Reply:
I agree that in general, it's bad practice to look at the "text representation" of error in tests as it is subject to change and becomes irritating during refactors. However, without checking what kind of error was returned you can't really tell if the test works as intended - it might return a different error because some other input data is wrong - how can you tell?

Reply:
It's not necessary to know (it's not necessary to ascertain which code path of the implementation is being executed), as long as for all invalid inputs an error is returned and for valid inputs no error is returned, then the component is functioning correctly.

So for testing a validation function I would first check that given valid input no error is returned, and then take the valid input and invalidate one field and call the validation function again to see that an error is returned and repeat that for all fields. Then I know that the validation function is working as expected, without having to put any further restrictions on its implementation.

It's true that maybe the error is wrong (has the wrong wording or is referencing some other problem) but similarly to how people don't generally test logging output, I don't think its particularly helpful to check errors at that level of detail.